### PR TITLE
added aten.log_normal_ decomp

### LINKED ISSUE #91275: [RFC] [Inductor] decompose aten.log_normal_
### 🚀 The feature, motivation and pitch

Currently aten.log_normal_ fallbacks to `FallbackKernel: torch.ops.aten.log_normal_.default`.

```python
import torch

def foo(x):
    return x.log_normal_()
    
opt_foo = torch.compile(foo)
x = torch.randn(100)
print(opt_foo(x))
```

`aten.log_normal_` is decomposable using `torch.randn` to something like:

```python
def log_normal(mean=1, std=2):
    return torch.exp(std * torch.randn(1) + mean)


x = []
for _ in range(10000):
    x.append(log_normal())

print(torch.mean(torch.tensor(x))) # verify E[X] is around e^(mean + 0.5 * std**2) ~= 20
print(torch.mean(torch.randn(10000).log_normal_())) # verify E[X] is around  e^(mean + 0.5 * std**2) ~= 20
```

### Correctness
```
X ~ Normal(mean, std**2), then
exp(X) ~ LogNormal(mean, std**2)
```

### Alternatives

_No response_

### Additional context

_No response_

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @mlazos @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire @fritzo @neerajprad @alicanb @nikitaved

### PR DESCRIPTION
**Pull request by min-jean-cho**
Fixes #91275

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_refs/__init__.py
Comment (min-jean-cho) [inline torch/_refs/__init__.py:5098]: I wonder if special care is needed to handle overflow, underflow for `log_normal`. For fp32, `torch.exp(x)` overflows for `x >= 89`. 

`torch.randn` practically won't cause this overflow as 99% of values will be in [-3, 3].
But for some `mean`, `std`, this will cause overflow.  
```
>>> torch.empty(100000).log_normal_(mean=89).max()
tensor(inf)
```
I wonder if this behavior is expected in `log_normal_` eager mode.

```
>>> np.random.lognormal(89, 1, 100000).max()
2.9258184949413217e+40
```

Comment (lezcano) [reply torch/_refs/__init__.py:5098]: I think this is alright, but thank you for bringing it up!

**Review by lezcano — COMMENTED** [review]
Could you add the relevant OpInfo for this operation as well? It won't be good enough to test the correctness of the implementation, but it will test that the type casting, the out variant and so on are there.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Review by lezcano — APPROVED** [review]
LGTM provided the tests pass!

**Comment by min-jean-cho** [comment]
[Log-normal] Hi all, thank you for the review. I see all tests have passed now - I've made few minor changes since the last approvals, could you help have a final scan ?

**Comment by min-jean-cho** [comment]
I will go ahead and merge this as the minor changes after the last approvals are necessary (same changes as https://github.com/pytorch/pytorch/pull/92047)

**Pull request closed without merging**
